int32 flagtype
int32 command_number
